terraform/node_resource_abstract.go

```diff
@@ -238,21 +238,29 @@ func (n *NodeAbstractResourceInstance) References() []*addrs.Reference {
 			// need to do a little work here to massage this to the form we now
 			// want.
 			var result []*addrs.Reference
-			for _, addr := range s.Current.Dependencies {
-				if addr == nil {
-					// Should never happen; indicates a bug in the state loader
-					panic(fmt.Sprintf("dependencies for current object on %s contains nil address", n.ResourceInstanceAddr()))
-				}
 
-				// This is a little weird: we need to manufacture an addrs.Reference
-				// with a fake range here because the state isn't something we can
-				// make source references into.
-				result = append(result, &addrs.Reference{
-					Subject: addr,
-					SourceRange: tfdiags.SourceRange{
-						Filename: "(state file)",
-					},
-				})
+			// It is (apparently) possible for s.Current to be nil. This proved
+			// difficult to reproduce, so we will fix the symptom here and hope
+			// to find the root cause another time.
+			//
+			// https://github.com/hashicorp/terraform/issues/21407
+			if s.Current != nil {
```

Comment:
In cases like this it might be beneficial to log the situation, in case we encounter it again and it leads us to a reproducible example. (plus you can keep the happy path further to the left)

Reply:
Oh that's an excellent suggestion, thank you. What do you prefer, `[TRACE]`, `[WARN]`?

Reply:
I think `WARN` would be appropriate for something we want to be alerted about if it arises.